reactive/kubernetes_master.py

```diff
@@ -2575,3 +2576,77 @@ def add_systemd_iptables_patch():
 
     # enable and run the service
     service_resume(service_name)
+
+
+@when('leadership.is_leader',
+      'kubernetes-master.components.started',
+      'endpoint.prometheus.joined',
+      'certificates.ca.available')
+def register_prometheus_jobs():
+    prometheus = endpoint_from_flag('endpoint.prometheus.joined')
+    # tls = endpoint_from_flag('certificates.ca.available')
+    loadbalancer = endpoint_from_flag('loadbalancer.available')
+    forced_lb_ips = hookenv.config('loadbalancer-ips').split()
+    hacluster_vip = get_hacluster_ip_or_hostname()
+
+    if forced_lb_ips:
```

Comment:
I wonder if we're to the point of moving this into a function. This is copied from somewhere else in here right? The logic for "get my api server endpoint" is getting complex and I would hate to end up with a bug only in one path.

Reply:
Yeah, I started to break this out into a function but then noticed that the code-paths were already slightly different and neither seemed exactly correct.  I think what I have here is the simplest and most correct of the 3(?), so if you're ok with that, I'll replace the others with this as a function.